spec/services/purchase_service_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe PurchaseService do
+
+  describe 'purchase' do
+    let!(:user) { create(:user) }
+    let(:address) do { 
+      address_1: 'Av bla',
+      address_2: 'Av Loop',
+      city: 'Porto',
+      state: 'SP',
+      country: 'Brazil',
+      zip: '2344567'
+    }
+    end
+    context 'when choose a invalid gateway' do      
+      let!(:cart) { create(:cart, user: user) }
+
+      it 'should returns a message error' do
+      result = PurchaseService.purchase(user, cart.id, address, nil)
+      expect(result).to eq( { errors: [{ message: 'Gateway not supported!' }], completed: false } ) 
```

Comment:
Como aqui o `it` está afirmando que retorna uma mensagem de erro, não faria mais sentido verificar _apenas_ a mensagem? ('Gateway not supported!')

Reply:
acho que faltou um tab aqui também :nail_care: